xbmc/rendering/dx/ScreenshotSurfaceWindows.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ScreenshotSurfaceWindows.h"
+
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "rendering/dx/DeviceResources.h"
+#include "ServiceBroker.h"
+#include "threads/SingleLock.h"
+#include "utils/log.h"
+#include "windowing/GraphicContext.h"
+
+#include "system_gl.h"
+
+#include <wrl/client.h>
+
+using namespace Microsoft::WRL;
+
+bool CScreenshotSurfaceWindows::Capture()
+{
+  CWinSystemBase *winsystem = CServiceBroker::GetWinSystem();
+  if (!winsystem)
+    return false;
+
+  CGUIComponent *gui = CServiceBroker::GetGUI();
+  if (!gui)
+    return false;
+
+  CSingleLock lock(winsystem->GetGfxContext());
+  gui->GetWindowManager().Render();
+
+  auto deviceResources = DX::DeviceResources::Get();
+  deviceResources->FinishCommandList();
+
+  ComPtr<ID3D11DeviceContext> pImdContext = deviceResources->GetImmediateContext();
+  ComPtr<ID3D11Device> pDevice = deviceResources->GetD3DDevice();
+  CD3DTexture& backbuffer = deviceResources->GetBackBuffer();
+  if (!backbuffer.Get())
+    return false;
+
+  D3D11_TEXTURE2D_DESC desc = { 0 };
+  backbuffer.GetDesc(&desc);
+  desc.Usage = D3D11_USAGE_STAGING;
+  desc.CPUAccessFlags = D3D11_CPU_ACCESS_READ;
+  desc.BindFlags = 0;
+
+  ComPtr<ID3D11Texture2D> pCopyTexture = nullptr;
```

Comment:
You don't need to initialise with `nullptr` here - that's the default.

Reply:
same